Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.DynamicExpression.cs

```diff
@@ -49,34 +51,39 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public DynamicExpression Update (IEnumerable<Expression> arguments)
     {
+      Contract.Ensures(Contract.Result<DynamicExpression>() != null);
```

Comment:
Yeah, this was a great source of confusion to me. It appears that as a general rule they promote nulls to empty collections, unless that collection is expected to be non-empty. If that's the case they require it be non-null.

There are many methods in Expression that don't require non-null on IEnumerable<> and params arrays for that reason. Should those be changed as well?